oak-run/src/main/java/org/apache/jackrabbit/oak/run/CreateGarbageCommand.java

```diff
@@ -0,0 +1,568 @@
+        }
+
+        return generateBasePaths;
+    }
+
+    private List<String> generateGarbageContinuously(CreateGarbageOptions options, Closer closer) throws Exception {
+
+        DocumentNodeStoreBuilder<?> builder = createDocumentMKBuilder(options, closer);
+        long startGenTimestamp = System.currentTimeMillis();
+
+        List<String> generatedGarbageBasePaths = new ArrayList<>();
+
+        int numberOfRuns = options.getNumberOfRuns();
+        int intervalSeconds = options.getGenerateIntervalSeconds();
+        Runnable task = () -> {
+            if (continuousRunIndex < numberOfRuns) {
+                try {
+                    String genBasePath = createGarbage(options, closer, continuousRunIndex, builder, startGenTimestamp);
+                    generatedGarbageBasePaths.add(genBasePath);
+                } catch (Exception e) {
+                    LOG.error("Error generating garbage in run " + continuousRunIndex, e);
+                }
+                LOG.info("Task executed. Count: " + (continuousRunIndex + 1));
+                continuousRunIndex++;
+            } else {
+                // Shutdown the executor once the task has run numberOfRuns times
+                continuousRunExecutor.shutdown();
+                LOG.info("Task completed " + numberOfRuns + " times. Stopping execution.");
+            }
+        };
+
+        // Schedule the task to run every intervalSeconds
+        continuousRunExecutor.scheduleAtFixedRate(task, 0, intervalSeconds, TimeUnit.SECONDS);
+
+        return generatedGarbageBasePaths;
+    }
+
+    /**
+     * Generate garbage nodes in the repository in order to allow for testing fullGC functionality.
+     *
+     * Returns the path of the generated GARBAGE_GEN_ROOT_PATH node (under the root).
+     * @param options
+     * @param closer
+     * @param runIndex
+     * @return
+     * @throws IOException
+     * @throws Exception
+     */
+    private String createGarbage(CreateGarbageOptions options, Closer closer, int runIndex) throws IOException, Exception {
+
+        DocumentNodeStoreBuilder<?> builder = createDocumentMKBuilder(options, closer);
+        long generationTimestamp = System.currentTimeMillis();
+
+        return createGarbage(options, closer, runIndex, builder, generationTimestamp);
+    }
+
+    private String createGarbage(CreateGarbageOptions options, Closer closer, int runIndex,
+                                 DocumentNodeStoreBuilder<?> builder, long timestamp) throws Exception {
+
+        if (builder == null) {
+            System.err.println("generateGarbage mode only available for DocumentNodeStore");
+            System.exit(1);
+        }
+
+        String generationBasePath = GEN_BASE_PATH + timestamp + "_" + runIndex;
+        documentNodeStore = builder.build();
+        closer.register(asCloseable(documentNodeStore));
+
+        NodeBuilder rootNode = documentNodeStore.getRoot().builder();
```

Comment:
nitpick : it could be somewhat confusing if a `NodeBuilder` is called `somethingNode`. Usually we call it `somethingBuilder` or `something` (as `somethingNode` is typically used for a `NodeState`)
(elsewhere I see you're also using the classic terminology, so this applies only to part of the code)

Reply:
Fixed, thank you for the comment.